data/mods/stadium/moves.ts

```diff
@@ -40,6 +40,22 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		inherit: true,
 		// FIXME: onBeforeMove() {},
 	},
+	haze: {
+		inherit: true,
+		onHit(target, source) {
+			this.add('-clearallboost');
+			for (const pokemon of this.getAllActive()) {
+				pokemon.clearBoosts();
+				// This should cure the status of both Pokemon, and subsequently recalculate stats to remove the Paralysis/Burn Speed Drop.
+				pokemon.cureStatus();
+				for (const id of Object.keys(pokemon.volatiles)) {
+					pokemon.removeVolatile(id);
+					this.add('-end', pokemon, id);
+				}
+				target.recalculateStats!();
```

Comment:
Should this be `pokemon` rather than `target`?

Reply:
you have a knack for waiting until its just too late to comment on things like this

Reply:
It should be both.

Reply:
yeah its in the for loop so it's `pokemon` is what urkerab was saying; I'll push a fix

Reply:
Sorry, I would have commented the first time I looked at this PR but I didn't see it then...

Reply:
> Sorry, I would have commented the first time I looked at this PR but I didn't see it then...

@urkerab I should clarify I was just joking, those kinds of catches are obviously always appreciated; my joke just didn't land like I had anticipated it would. Sorry!